### ISSUE TITLE
HTTP Support from Protocols - Http2 Spike

### ISSUE THREAD

**Issue by eperinan** [higherkindness/mu-scala#205]
This issue comes from https://github.com/frees-io/freestyle-rpc/issues/182.

After #203, we need to enable HTTP2 like this example https://github.com/http4s/http4s/blob/ecf32b9377997f17ed4c8858ec1471d0836d8522/examples/blaze/src/main/scala/com/example/http4s/blaze/BlazeHttp2Example.scala.

**Comment by L-Lavigne** [comment]
Just a quick note that HTTP/2 support with `.enableHttp2()` is ignored (with a logged warning) unless TLS is also enabled. I think this is done with `.withSSLContext()` but my quick testing with that wasn't conclusive. The `http4s` docs and issues seem a bit vague on the status of HTTP/2 support on the client side.

**Comment by juanpedromoreno** [comment]
I think this spike has been already done by @L-Lavigne . Thanks for doing it!

**Comment by L-Lavigne** [comment]
As discussed offline we'll keep this issue open for now, and find out if client-side HTTP/2 support in http4s/blaze is mature enough for us to use.

**Comment by juanpedromoreno** [comment]
This is probably a bit outdated. Closing it for now.
